Tallies/TallyMaps/Maps1D/radialMap_class.f90

```diff
@@ -0,0 +1,267 @@
+  use grid_class,         only : grid
+  use particle_class,     only : particleState
+  use outputFile_class,   only : outputFile
+  use tallyMap1D_inter,   only : tallyMap1D, kill_super => kill
+
+  implicit none
+  private
+
+  !!
+  !! Divides space into a radial mesh in cylindrical or spherical co-ordinates
+  !!
+  !! Interface:
+  !!   tallyMap1D interface
+  !!
+  !! NOTE:
+  !!   Behaviour of points exactly at the boundary of bins is undefined.
+  !!   particle can end-up in either of the two
+  !!
+  !! Sample Dictionary Input:
+  !!  radialMap {
+  !!    type radialMap;
+  !!    axis x;                 // Optional. Default is spherical map
+  !!    #origin (1.0 0.0 0.0);# // Optional. Default (0.0 0.0 0.0)
+  !!    grid lin;
+  !!    #min 2.0;#              // Optional. Default 0.0
+  !!    max  10.0;
+  !!    N 10;
+  !!    }
+  !!
+  type, public, extends (tallyMap1D) :: radialMap
+    private
+    real(defReal), dimension(:), allocatable     :: origin
+    integer(shortInt), dimension(:), allocatable :: axis
+    type(grid)         :: bounds
+    integer(shortInt)  :: N = 0
+
+  contains
+    ! Superclass
+    procedure :: init
+    procedure :: bins
+    procedure :: getAxisName
+    procedure :: map
+    procedure :: print
+    procedure :: kill
+
+  end type radialMap
+
+contains
+
+  !!
+  !! Initialise tallyMap from a dictionary
+  !!
+  !! See tallyMap for specification.
+  !!
+  subroutine init(self, dict)
+    class(radialMap), intent(inout)          :: self
+    class(dictionary), intent(in)            :: dict
+    real(defReal), dimension(:), allocatable :: grid
+    character(nameLen)                       :: type
+    real(defReal)                            :: min, max, vol, exp
+    integer(shortInt)                        :: i
+    logical(defBool)                         :: spherical
+    character(100), parameter :: Here = 'init (radialMap_class.f90)'
+
+    ! Check if the map is cylindrical or spherical, and orientation of the cylinder
+    call dict % getOrDefault(type, 'axis', 'xyz')
+    spherical = .false.
+
+    select case(type)
+      case('x')
+        allocate(self % axis(2), self % origin(2))
+        self % axis = [Y_AXIS, Z_AXIS]
+
+      case('y')
+        allocate(self % axis(2), self % origin(2))
+        self % axis = [X_AXIS, Z_AXIS]
+
+      case('z')
+        allocate(self % axis(2), self % origin(2))
+        self % axis = [X_AXIS, Y_AXIS]
+
+      case('xyz')
+        allocate(self % axis(3), self % origin(3))
+        self % axis = [X_AXIS, Y_AXIS, Z_AXIS]
+        spherical = .true.
+
+      case default
+        call fatalError(Here, 'Keyword orientation must be x, y, z or nothing for spherical. It is: '//type)
+
+    end select
+
+    ! Check & load origin
+    if (spherical) then
+      call dict % getOrDefault(self % origin, 'origin', [ZERO, ZERO, ZERO])
+    else
+      call dict % getOrDefault(self % origin, 'origin', [ZERO, ZERO])
+    end if
```

Comment:
I guess I was imitating what done in cylinderMap, where the origin only had two entries. I am changing both cylinderMap and radialMap to read origins with 3 entries to avoid confusion for the user, but then only the two relevant values are saved. In radialMap, where in the norm2 calculation one wants arbitrarily 2 or 3 values according to whether the map is spherical or cylindrical, having an allocatable origin is more convenient.